internal/cli/cmd/dac/setup/setup.go

```diff
@@ -0,0 +1,249 @@
+	if err != nil {
+		return err
+	}
+	defer gzipReader.Close()
+	tarReader := tar.NewReader(gzipReader)
+
+	// Extract each CUE dep to the destination path
+	// TODO simplify the code with https://github.com/mholt/archiver? Wait for stable release of v4 maybe
+	depsFolderFound := false
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+
+		// Remove the wrapping folder for following evaluations
+		currentDepPath := removeFirstFolder(header.Name)
+
+		if currentDepPath == depsFolderName {
+			depsFolderFound = true
+		}
+		if !strings.HasPrefix(currentDepPath, depsFolderName) {
+			continue
+		}
+
+		newDepPath := fmt.Sprintf("%s/%s", depsRootDstPath, currentDepPath)
+
+		switch header.Typeflag {
+		case tar.TypeDir:
+			if err := os.Mkdir(newDepPath, 0666); err != nil {
+				return fmt.Errorf("can't create dir %s: %v", newDepPath, err)
+			}
+			logrus.Debugf("dir %s created succesfully", newDepPath)
+		case tar.TypeReg:
+			outFile, err := os.Create(newDepPath)
+			if err != nil {
+				return fmt.Errorf("can't create file %s: %v", newDepPath, err)
+			}
+			defer outFile.Close()
+			if _, err := io.CopyN(outFile, tarReader, maxFileSize); err != nil {
+				return fmt.Errorf("can't copy content from %s: %v", header.Name, err)
+			}
+			logrus.Debugf("file %s extracted succesfully", newDepPath)
+		default:
+			return fmt.Errorf("unknown type: %b in %s", header.Typeflag, header.Name)
+		}
+	}
+
+	if !depsFolderFound {
+		return fmt.Errorf("CUE dependencies not found in archive")
+	}
+
+	return nil
+}
+
+func removeFirstFolder(filePath string) string {
+	separatorChar := "/" // force the usage of forward slash for strings comparison
+
+	// Split the path into individual components
+	components := strings.Split(filePath, separatorChar)
+
+	// Remove the top folder if there is at least one folder in the path
+	if len(components) > 1 {
+		components = components[1:]
+	}
+
+	// Join the components back into a path
+	resultPath := strings.Join(components, separatorChar)
+
+	return resultPath
+}
+
+func (o *option) SetWriter(writer io.Writer) {
+	o.writer = writer
+}
+
+func NewCMD() *cobra.Command {
+	o := &option{}
+	cmd := &cobra.Command{
+		Use:   "setup",
+		Short: "Sets up a local development environment to do Dashboard-as-Code",
+		Long: `
+takes care of adding the CUE sources from Perses as external dependencies to your DaC repo.
+/!\ It must be executed at the root of your repo.
+`,
+		Example: `
+# DaC setup when you are connected to a server
+percli dac setup
+
+# DaC setup when you are not connected to a server, you need to provide the Perses version to consider for dependencies retrieval
+percli dac setup --version 0.42.1
+`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return persesCMD.Run(o, cmd, args)
+		},
+	}
+	cmd.Flags().StringVar(&o.version, "version", "", "Version of Perses from which to retrieve the CUE dependencies.")
```

Comment:
I'm wondering if we should manage the setup for the go sdk as well here. That's kind of weird from my point of view the command `dac init` would support only one language

Reply:
IMHO better to keep it simple thus CUE-opinionated for now, and to refactor with a next PR when we'd want to make it work with the Go SDK, but that's a matter of preference I guess.